### DISCUSSION TITLE
Plans to support Kademlia DHT?

### DISCUSSION THREAD

**Discussion by polyhobbyist** [NethermindEth/dotnet-libp2p#65] [Q&A]
Hello there,
Question: 
Are there plans to support, or specifically not support Kademlia DHT? 

Discussion:
I see that Discv5 is specifically called out and interested in being supported, with a help wanted tag; but I see no mention of Kademila. I wanted to understand if this was specifically by design, an oversight in the roadmap or lack of resources.
(If the latter, I might be interested in building it)

Thank you for working on this project!

**Comment by flcl42** [comment]
@polyhobbyist hello! The main focus right now is stabilization and gossipsub completion. But Kademilia is on the list of long term goals and it'd be awesome to have it at some point! We will be glad to help with any questions and improvements required for it to work, in case you decide to implement it.

**Reply by polyhobbyist** [reply]
Thank you for confirming! I've created the issue https://github.com/NethermindEth/dotnet-libp2p/issues/67 to implement it. Once I complete my current workstream using libp2p, I'll investigate implementing this.

**Comment by flcl42** [comment]
It's supported now
